Em app policy decode and encoder

Nikita-Hakai opened the pull request.

vivianecordeiro-sky commented on include/wifi_base.h:
```diff
@@ -1066,6 +1066,57 @@ typedef struct {
+
+typedef struct {
+    int interval;
+    marker_name managed_client_marker;
```
should this be a array?

Nikita-Hakai replied:
Since its a string value, and I don't see usage of dynamic declaration for similar cases. Hence kept as an array.

vivianecordeiro-sky commented:
Hi  @Nikita-Hakai, in general the code is going in the right direction.
onewifi and rdk-wifi-hal repos, both uses clang. Please use the tool to format your code. Please only use in the new code added since not all code is following the format at this moment.

Nikita-Hakai commented on source/webconfig/wifi_decoder.c:
```diff
@@ -5159,3 +5159,110 @@ webconfig_error_t decode_radio_temperature_stats_object(wifi_provider_response_t
+    if (ap_metrics_policy != NULL) {
+        decode_param_integer(ap_metrics_policy, "Interval", param);
+        em_config->ap_metric_policy.interval = param->valuedouble;
+        printf("    AP Metrics Reporting Interval: %f\n", em_config->ap_metric_policy.interval);
```
Addressed

Nikita-Hakai commented on source/webconfig/wifi_decoder.c:
```diff
@@ -5159,3 +5159,110 @@ webconfig_error_t decode_radio_temperature_stats_object(wifi_provider_response_t
     return webconfig_error_none;
 }
+
+webconfig_error_t decode_em_policy_object(const cJSON *em_cfg, em_config_t *em_config)
```
Addressed

Nikita-Hakai commented:
> Hi @Nikita-Hakai, in general the code is going in the right direction. onewifi and rdk-wifi-hal repos, both uses clang. Please use the tool to format your code. Please only use in the new code added since not all code is following the format at this moment.

Hi @vivianecordeiro-sky, I addressed the comments. Kindly check if its okay to be merged now.
